### DISCUSSION TITLE
Get historical data grouped by day

### DISCUSSION THREAD

**Discussion by franciscop** [prisma/prisma#12233] [Q&A]
So I want to get 7 rows, from today (included) back in time, one for each day. [This feature request](https://github.com/prisma/prisma/issues/6653) would be perfect, but from the lack of it I'm following [this other way](https://github.com/prisma/prisma-client-js/issues/419#issuecomment-656015435). However, it results in monstrous code:

```js
// Get each day's usage information
const findUsage = async (userId, start, end) => {
  const transferAggregate = await db.transfer.aggregate({
    where: { userId, created: { gte: start, lt: end } },
    _sum: { size: true, },
  });
  const fileAggregate = await db.file.aggregate({
    where: { userId, created: { gte: start, lt: end } },
    _sum: { size: true, },
  });
  const operationsAggregate = await db.operation.aggregate({
    where: { userId, created: { gte: start, lt: end } },
    _sum: { total: true, },
  });
  // ...
};

// Make the date for today, minus the given number of days, all at 00:00:00
const toDateDiff = (num) => {
  const d = new Date(new Date().setDate(new Date().getDate() - num));
  d.setUTCHours(0, 0, 0, 0);
  return new Date(d);
};

// Main function to get them all
const getHistory = async (userId) => {
  return await Promise.all(
    [0, 1, 2, 3, 4, 5, 6]
      .map((diff) => [toDateDiff(diff), toDateDiff(diff - 1)])
      .reverse()
      .map((times) => findUsage(userId, ...times))
  );
}
```

It *works* and surprisingly for me it's decently fast (~60ms). However, this is 21 (7 * 3) queries for every time the user visits their dashboard, and right now the DB is empty so I am very worried that it'll blow up with any kind of meaningful data. My question is: **with *prisma*, not going down to SQL, do you see any way that I could improve these queries?** I could make `findUsage` parallel but I'm more concerned if I could somewhat join those 3 queries into 1, or somehow group the 7-days-loop.

**Comment by pantharshit00** [comment]
Hello [franciscop](https://github.com/franciscop), our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. It's very important for us to address your question, so let us know in a comment if this is still an open discussion. 

This above code is good here. As you have said, you can combine the three queries into one and make them run at the same time.

**Comment by franciscop** [comment]
> "you can combine the three queries into one"

Sorry but how? I was asking if this is possible, not saying that I know how to do that.

The linked related issue is still open:
https://github.com/prisma/prisma/issues/6653

**Reply by pantharshit00** [reply]
I am sorry I misread the description at first. You will need to wait for that feature implementation. I will convey the feedback to the product team and will reference the open issue.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
